arches/app/src/components/ControlledListManager/EditLabel.vue

```diff
@@ -0,0 +1,167 @@
+
+const value = ref(props.label.value);
+const language = ref(props.label.language);
+
+const visible = defineModel<boolean>({ required: true });
+
+const languageOptions = computed(() => {
+    if (!props.languageMap) {
+        return [];
+    }
+    return (
+        Object.entries(props.languageMap).map(([code, label]) => {
+            return {
+                label,
+                code,
+            };
+        })
+    );
+});
+
+const toast = useToast();
+const { $gettext } = useGettext();
+const staticItemLabel = $gettext("Item Label");
+const staticLanguageLabel = $gettext("Language");
+
+const onSave = async () => {
+    const upsertedLabel = await upsertLabel(
+        {
+            id: props.label.id,
+            value: value.value,
+            language: language.value,
+            valuetype: props.label.valuetype,
+            itemId: props.item.id,
+        },
+        toast,
+        $gettext,
+    );
+
+    if (upsertedLabel) {
+        if (props.onInsert !== null) {
+            props.onInsert(upsertedLabel);
+            value.value = "";
+        } else {
+            /* eslint-disable vue/no-mutating-props */
+            props.label.language = language.value;
+            props.label.value = value.value;
+            /* eslint-enable vue/no-mutating-props */
+        }
+        visible.value = false;
+    }
+};
+</script>
+
+<template>
+    <Dialog
+        v-model:visible="visible"
+        modal
+        :draggable="false"
+        :header
+        :style="{ width: '40rem' }"
+        :pt="{
+            header: { style: { backgroundColor: '#2d3c4b' } },
+            title: { style: { color: 'white' } },
+            closeButtonIcon: { style: { color: 'white' } },
+        }"
+    >
+        <div class="form-input">
+            <label for="value">{{ staticItemLabel }}</label>
+            <InputText
+                id="value"
+                v-model="value"
+                autocomplete="off"
+            />
+        </div>
+        <div class="form-input">
+            <span
+                id="language-label"
+                style="margin-bottom: 5px"
+            >{{ staticLanguageLabel }}</span>
+            <Dropdown
+                v-model="language"
+                aria-labelledby="language-label"
+                :options="languageOptions"
+                option-label="label"
+                option-value="code"
+                :pt="{
+                    input: { style: { fontSize: 'small' } },
+                    panel: { style: { fontSize: 'small' } },
+                }"
+            />
+        </div>
+        <div class="controls">
+            <Button
+                type="button"
+                :label="arches.translations.save"
+                :disabled="!value || !language"
+                @click="onSave"
+            />
+            <Button
+                type="button"
+                :label="arches.translations.cancelEdit"
```

Comment:
we're trying to move i18n out of the `arches` object. This and any other references should be inlined.

Reply:
I'd love more input on this one, esp. from @apeters. A few considerations led me down this path:
- The translation coverage of arches will immediately degrade if we add new strings and don't translate them (assuming the extracted messages are not de-duplicated against the old strings, but my guess is that they're not--I'd love to know for sure)
- Even if they are translated, it creates unnecessary work for translators
- Even if they are translated, it creates the risk of "drift" where "Cancel edit" gets translated two different ways and creates inconsistent UX. More important for strings a little more interesting than "cancel edit" to be sure...
- `arches.translations.cancelEdit` is not going away until all the knockout is gone, and it's just an easy find-n-replace when that happens

Thoughts?

Reply:
These are all reasonable assumptions. I suppose since `arches.translations` will exist until the cutover is complete it doesn't super matter whether the translations are moved over piecemeal or all together.

I arrived at my conclusion by the heuristic that we don't want to _add_ strings to `arches.translations`, and we don't want to have mixed translation logic in new components. This leaves moving all strings to the new pattern.

However I'm happy to defer to @apeters on this 👍

Reply:
Hey @jacobtylerwalls and @chrabyrd.  I took a long hard look at this and I think it would be nice to have all these new Vue files use our latest patterns and thinking.  Which means moving away from using the `arches` object even in light of the arguments @jacobtylerwalls posed above.

The bottom line is that :
- our translations are already horribly out of date and will require a lot of work to bring back in sync
- there are gnu gettext utils that can assist in managing these translation files (merging, etc..) which we will have to do in either case
- if we did opt for the search and replace method, we'd still want to import `gettext` in every file that needs translations even if we're only using the `arches` object (eg: `SearchAndDelete.vue`) 

If we do what I'm suggesting, then we can be assured that these files won't have to be touched when we finally tackle the `javascript.htm` issue and i18n in general.

Reply:
Thanks, very helpful context. Removed arches.translations in 5560ccf